staging/src/k8s.io/dynamic-resource-allocation/structured/allocator_test.go

```diff
@@ -165,6 +172,24 @@ func request(name, class string, count int64, selectors ...resourceapi.DeviceSel
 	}
 }
 
+func subRequest(name, class string, count int64, selectors ...resourceapi.DeviceSelector) resourceapi.DeviceSubRequest {
+	return resourceapi.DeviceSubRequest{
+		Name:            name,
+		Count:           count,
+		AllocationMode:  resourceapi.DeviceAllocationModeExactCount,
```

Comment:
We should have some tests with subrequest with All mode. And perhaps even one where we:
- have a request that asks for 1 classA
- have another request with subrequests that ask for
  - ALL classA (should not be satisfied, since first request will have taken one)
  - one classA (should be satsified)

Eventually if we add scoring the specific result may change (depends on scoring algo), but for now it should be fixed as above (I think).

Reply:
There were already tests with All mode and added a test with the specific scenario you describe here.